core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -177,7 +177,11 @@ public void setCompression(String compressorName) {
     checkState(!sendHeadersCalled, "sendHeaders has been called");
 
     compressor = compressorRegistry.lookupCompressor(compressorName);
-    checkArgument(compressor != null, "Unable to find compressor by name %s", compressorName);
+    if (compressor == null) {
+      throw Status.INTERNAL
```

Comment:
The problem is that it shouldn't be a Status, here. No RPC failed. Instead, maybe the application should catch this exception and turn it into a Status.